pkg/controller/combo_controller.go

```diff
@@ -2,36 +2,161 @@ package controller
 	log := c.log.WithValues("request", req)
-	log.Info("reconciling combination")
 
-	in := &v1alpha1.Combination{}
-	if err := c.Get(ctx, req.NamespacedName, in); err != nil {
+	log.V(1).Info("new combination event inbound")
+
+	// Attempt to retrieve the requested combination CR
+	combination := &v1alpha1.Combination{}
+	err := c.Get(ctx, req.NamespacedName, combination)
+	if err != nil {
 		return reconcile.Result{}, client.IgnoreNotFound(err)
 	}
 
-	return reconcile.Result{}, nil
+	// If combination is being deleted, remove from queue
+	if !combination.ObjectMeta.DeletionTimestamp.IsZero() {
+		log.Info("combination is being deleted, ignoring event")
+		return reconcile.Result{}, nil
+	}
+
+	// Attempt to retrieve the template referenced in the combination CR
+	template := &v1alpha1.Template{}
+	if err := c.Get(ctx, types.NamespacedName{Name: combination.Spec.Template}, template); err != nil {
+		combination.SetCondition(metav1.Condition{
+			Type:               v1alpha1.TypeInvalid,
+			Status:             metav1.ConditionFalse,
+			Reason:             v1alpha1.ReasonTemplateNotFound,
+			LastTransitionTime: metav1.NewTime(time.Now()),
+			Message:            fmt.Sprintf("failed to retrieve %s template: %s", combination.Spec.Template, err.Error()),
+		})
+		return reconcile.Result{}, errors.NewAggregate([]error{err, c.Status().Update(ctx, combination)})
+	}
+
+	// Build combination stream to be utilized in template builder
+	comboStream := combinationPkg.NewStream(
+		combinationPkg.WithArgs(formatArguments(combination.Spec.Arguments)),
+		combinationPkg.WithSolveAhead(),
+	)
+
+	// Create a new template builder
+	builder, err := templatePkg.NewBuilder(strings.NewReader(template.Spec.Body), comboStream)
+	if err != nil {
+		combination.SetCondition(metav1.Condition{
+			Type:               v1alpha1.TypeInvalid,
+			Status:             metav1.ConditionFalse,
+			Reason:             v1alpha1.ReasonTemplateBodyInvalid,
+			LastTransitionTime: metav1.NewTime(time.Now()),
+			Message:            fmt.Sprintf("failed to construct a builder out of %s template body:  %s", combination.Spec.Template, err.Error()),
+		})
+		return reconcile.Result{}, errors.NewAggregate([]error{err, c.Status().Update(ctx, combination)})
+	}
+
+	// Build the manifest combinations
+	generatedManifests, err := builder.Build(ctx)
+	if err != nil {
+		combination.SetCondition(metav1.Condition{
+			Type:               v1alpha1.TypeInvalid,
+			Status:             metav1.ConditionFalse,
+			Reason:             v1alpha1.ReasonEvaluationsInvalid,
+			LastTransitionTime: metav1.NewTime(time.Now()),
+			Message:            fmt.Sprintf("failed to generate manifest %s combinations: %s", combination.Spec.Template, err.Error()),
+		})
+		return reconcile.Result{}, errors.NewAggregate([]error{err, c.Status().Update(ctx, combination)})
+	}
+
+	log.Info(fmt.Sprintf("reconciliation of %s combination successful!", combination.Name))
```

Comment:
I don't think that we need to print on success, this can be found in the status of the combination.